src/code/sched.c

```diff
@@ -1,9 +1,61 @@
 #include <ultra64.h>
 #include <global.h>
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/sched/func_800C82A0.s")
+#define RSP_DONE_MSG 667
+#define RDP_DONE_MSG 668
+#define ENTRY_MSG 670
 
-void func_800C84E4(SchedContext* sc, UNK_TYPE arg1) {
+#define ENSURE_PTR(ptr, member) (ptr != NULL ? ptr->member : NULL)
```

Comment:
Personally not a fan of this macro since it's there are essentially just as many places where it can't be used as where it can (only 3 instances, one of which would be completely fine as a ternary, and the other 2 wouldn't be that bad).